internal/queue/cache/memory_cache.go

```diff
@@ -252,31 +260,40 @@ func (cache *MemoryCache) PullMessages(ctx context.Context, queue string, n int6
 
 	result := make([]string, 0, utils.MinInt64(n, total))
 
-	// TODO implement score filter
-	//now := time.Now().Unix()
-	//
-	//score := int64(0)
-	//if scoreFilter > 0 {
-	//	score = now - scoreFilter
-	//}
-
+	filteredOut := make([]*MemoryMessageEntry, 0)
 	for i := int64(0); i < n && i < total && cache.queues[queue].Len() != 0; i++ {
 		element := cache.queues[queue].Front()
 
 		entry := element.Value.(*MemoryMessageEntry)
 
 		cache.queues[queue], _ = removeEntry(entry, cache.queues[queue])
 
+		if minScore != nil && entry.score < *minScore {
+			filteredOut = append(filteredOut, entry)
+
+			continue
+		}
+
+		if maxScore != nil && entry.score > *maxScore {
+			filteredOut = append(filteredOut, entry)
+
+			continue
+		}
```

Comment:
Maybe this could all belong to a single if statement in order to reduce the code duplicated from both ifs